### PR TITLE
Adding documentation for FM and COOP/COEP

### PR DESCRIPTION
**Pull request by eozmen410**
Moving our changes to your repo so you can later submit a PR upstream!

### CODE REVIEW

File: source/core-developers/coep-interceptor.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+COEP is now supported by all major browsers.
+
+
+
+[More information about COEP](https://web.dev/why-coop-coep/#coep).
+
+## Parameters
+
+- `exemptedPaths` - Set of opt out endpoints that are meant" to serve cross-site traffic. Paths should either be relative or absolute. No default value for field.
```
Comment (salcho) [inline source/core-developers/coep-interceptor.md]: Please see comment on `exemptedPaths` for coop-interceptor.

```suggestion
- `exemptedPaths` - Set of opt out endpoints that are meant" to serve cross-site traffic. Paths should either be relative or absolute. This field is empty by default.
```

File: source/core-developers/coop-interceptor.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+COOP is now supported by all major browsers.
+
+
+[More information about COOP](https://web.dev/why-coop-coep/#coop).
+
+## Parameters
+
+- `exemptedPaths` - Set of opt out endpoints that are meant to serve cross-site traffic. Paths should either be relative or absolute.
+- `mode` - The policy mode COOP should followe
```
Comment (salcho) [inline source/core-developers/coop-interceptor.md]: Please provide the list of available modes here

```suggestion
- `mode` - The policy mode COOP should follow
```

File: source/core-developers/coop-interceptor.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+The COOP header can have one of 3 values: `same-origin`, `same-origin-allow-popups`, `unsafe-none`.  If the COOP values are the same, and the origins of the documents match the relationship declared in the COOP header value, documents can interact with each other. Otherwise if at least one of the documents sets COOP, the browser will create a new browsing context group severing the link between the documents. Sites can use `same-origin-allow-popups` to allow popups they open to be in their browsing context group (unless the popup's own COOP prevents this).
+
+COOP is now supported by all major browsers.
+
+
+[More information about COOP](https://web.dev/why-coop-coep/#coop).
+
+## Parameters
+
+- `exemptedPaths` - Set of opt out endpoints that are meant to serve cross-site traffic. Paths should either be relative or absolute.
```
Comment (salcho) [inline source/core-developers/coop-interceptor.md]: I don't think these paths can be absolute. We [only compare the context path field of incoming requests](https://github.com/apache/struts/blob/889601522878c8fd81493e79b5d47778f41f0e67/core/src/main/java/org/apache/struts2/interceptor/CoopInterceptor.java#L64) and there [is no unit test](https://github.com/apache/struts/blob/889601522878c8fd81493e79b5d47778f41f0e67/core/src/test/java/org/apache/struts2/interceptor/CoopInterceptorTest.java#L50) documenting this behaviour. If this is correct, let's please say that paths should contain leading slashes and must be relative.

File: source/core-developers/fetch-metadata-interceptor.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+Sec-Fetch-Site == 'cross-site' AND (Sec-Fetch-Mode != 'navigate'/'nested-navigate' OR method NOT IN [GET, HEAD])
+```
+
+Refer to [Implementing a Resource Isolation Policy](https://web.dev/fetch-metadata/#implementing-a-resource-isolation-policy) for further information on implementing effective Resource Isolation Policies.
+Fetch Metadata is supported in all major browsers
+
+## Parameters
+
+- `exemptedPaths` - Set of opt out endpoints that are meant to serve cross-site traffic. Paths should either be relative or absolute. No default value for field.
+- `resourceIsolationPolicy` - Instance of Resource Isolation Policy class implementing the logic for the requests filtering. The default value for this field is a new `DefaultIsolationPolicy`.
```
Comment (salcho) [inline source/core-developers/fetch-metadata-interceptor.md]: This isn't an actual parameter of the interceptor. As a matter of fact, the resource isolation policy cannot be configured at the moment (apart from subclassing the interceptor, perhaps). The [only parameter this interceptor receives from XML config](https://github.com/apache/struts/blob/54b1568a5041f456f98266af7897585b861accc8/core/src/main/java/org/apache/struts2/interceptor/FetchMetadataInterceptor.java#L56) is `exemptedPaths`
```suggestion
```

File: source/core-developers/interceptors.md [hunk 5]
```diff
@@ -116,15 +116,18 @@ specified in the `<interceptors/>` tag.
...
 |[Create Session Interceptor](create-session-interceptor.html)|createSession|Create an HttpSession automatically, useful with certain Interceptors that require a HttpSession to work properly (like the TokenInterceptor)|
 |[Clear Session Interceptor](clear-session-interceptor.html)|clearSession|This interceptor clears the HttpSession.|
 |[Debugging Interceptor](debugging-interceptor.html)|debugging|Provides several different debugging screens to provide insight into the data behind the page.|
 |[Default Workflow Interceptor](default-workflow-interceptor.html)|workflow|Calls the validate method in your Action class. If Action errors are created then it returns the INPUT view.|
 |[Exception Interceptor](exception-interceptor.html)|exception|Maps exceptions to a result.|
 |[Execute and Wait Interceptor](execute-and-wait-interceptor.html)|execAndWait|Executes the Action in the background and then sends the user off to an intermediate waiting page.|
+|[Fetch Metadata Interceptor](fetch-metadata-interceptor.html)|fetchMetadata|Implements the Fetch Metadata policy on incoming requests used to protect against CSRF, XSSI, and cross-origin information leaks.|
```
Comment (salcho) [inline source/core-developers/interceptors.md]:
```suggestion
|[Fetch Metadata Interceptor](fetch-metadata-interceptor.html)|fetchMetadata|Implements Resource Isolation Policies on incoming requests used to protect against CSRF, XSSI, and cross-origin information leaks.|
```

File: source/core-developers/interceptors.md [hunk 6]
```diff
@@ -116,15 +116,18 @@ specified in the `<interceptors/>` tag.
 |[Annotation Workflow Interceptor](annotation-workflow-interceptor.html)|annotationWorkflow|Invokes any annotated methods on the action.|
 |[Chaining Interceptor](chaining-interceptor.html)|chain|Makes the previous Action's properties available to the current Action. Commonly used together with <result type="chain"> (in the previous Action).|
 |[Checckbox Interceptor](checkbox-interceptor.html)|checkbox|Adds automatic checkbox handling code that detect an unchecked checkbox and add it as a parameter with a default (usually 'false') value. Uses a specially named hidden field to detect unsubmitted checkboxes. The default unchecked value is overridable for non-boolean value'd checkboxes.|
+|[COEP Interceptor](coep-interceptor.html)|coep|Implements the COEP policy on incoming requests used to protect a document from loading any non-same-origin resources which don't explicitly grant the document permission to be loaded.|
```
Comment (salcho) [inline source/core-developers/interceptors.md]:
```suggestion
|[COEP Interceptor](coep-interceptor.html)|coep|Implements the Cross-Origin Embedder Policy on incoming requests used to protect a document from loading any non-same-origin resources which don't explicitly grant the document permission to be loaded.|
```

File: source/core-developers/interceptors.md [hunk 7]
```diff
@@ -116,15 +116,18 @@ specified in the `<interceptors/>` tag.
...
 |[Conversion Error Interceptor](conversion-error-interceptor.html)|conversionError|Adds conversion errors from the ActionContext to the Action's field errors|
 |[Cookie Interceptor](cookie-interceptor.html)|cookie|Inject cookie with a certain configurable name / value into action. (Since 2.0.7.)|
 |[Cookie Provider Interceptor](cookie-provider-interceptor.html)|cookieProvider|Transfer cookies from action to response (Since 2.3.15.)|
+|[COOP Interceptor](coop-interceptor.html)|coop|Implements the COOP policy on incoming requests used to isolate resources against side-channel attacks and information leaks.|
```
Comment (salcho) [inline source/core-developers/interceptors.md]:
```suggestion
|[COOP Interceptor](coop-interceptor.html)|coop|Implements the Cross-Origin Opener Policy on incoming requests used to isolate resources against side-channel attacks and information leaks.|
```

**Review by salcho — CHANGES_REQUESTED** [review]
In general: `exemptedPaths` is a parameter that does exact string matching against the path associated with URL requests. That means that the parameter should **only receive relative paths with a trailing slash**. All parameters have default values, so instead of saying 'this parameter has no default value' we must say 'the default set of exempted paths is empty'. This applies for all the files in this PR.

Additionally, there seems to be a confusion between Fetch Metadata and Resource Isolation Policies. Fetch Metadata specifies what the Sec-Fetch-* headers are and what they mean. Resource Isolation Policies read those headers and take actions depending on their values. This means we should never say "here's how to implement Fetch Metadata" (because only browsers implement that), instead we should say "here's how you can implement a resource isolation policy based on Fetch Metadata headers".

File: source/core-developers/coop-interceptor.md (hunk 3 above)
Comment (gchatz22) [reply source/core-developers/coop-interceptor.md]: Correct, I had in mind the reporting URL of CSP that can be absolute and confused myself.

**Review by gchatz22 — COMMENTED** [review]
Thanks Sal!

File: source/core-developers/coep-interceptor.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+COEP is now supported by all major browsers.
+
+
+
+[More information about COEP](https://web.dev/why-coop-coep/#coep).
+
+## Parameters
+
+- `exemptedPaths` - Set of opt out endpoints that are meant" to serve cross-site traffic. Paths should contain leading slashes and must be relative. This field is empty by default.
```
Comment (salcho) [inline source/core-developers/coep-interceptor.md]:
```suggestion
- `exemptedPaths` - Set of opt out endpoints that are meant to serve cross-site traffic. Paths should contain leading slashes and must be relative. This field is empty by default.
```

File: source/core-developers/coep-interceptor.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+- `enforcingMode` - Boolean variable allowing the user to let COEP operate in `enforcing`, which blocks both resource and reports violation, or `report-only` mode, which only reports violation. Default value for field is `false`.
```
Comment (salcho) [inline source/core-developers/coep-interceptor.md]:
```suggestion
- `enforcingMode` - Boolean variable allowing the user to let COEP operate in `enforcing`, which blocks both resource and reports violations, or `report-only` mode, which only reports violations. Default value for field is `false`.
```

**Review by salcho — CHANGES_REQUESTED** [review]
amost there!

**Review by salcho — APPROVED** [review]
LGTM! 🥇

**Pull request merged**